### ISSUE TITLE
Per-step gate offset

### ISSUE THREAD

**Issue by hotelbachelard** [westlicht/performer#253]
Hi, 

I'd love to be able to do a negative gate offset on a step. Currently you can only go between 0 and 87. 5%, but you can't make the step trigger earlier. The screen also puts the 0% mark in the middle, so I assumed that you can go to negative values. 

Many thanks!

**Comment by vafu** [comment]
Hi, I have a PR with negative offsets implemented and I need some help testing it, would you mind trying my changes out?
https://github.com/westlicht/performer/pull/234

**Comment by hotelbachelard** [comment]
Hey there,
Sure! Thanks for asking

---------------Music: http://www.bachelard.bandcamp.com

 

    On Monday, November 30, 2020, 8:13:34 p.m. EST, vafu <notifications@github.com> wrote:  
 
 


Hi, I have a PR with negative offsets implemented and I need some help testing it, would you mind trying my changes out?
#234

**Comment by hotelbachelard** [comment]
Hey @vafu - your latest version has a glitch. In the Step mode/page, normally, you can only turn on/off a step gate when you're on the Gate menu/page (1st tab, 1st item), and you need to press on the step + encoder when you're on any other  menu (Retrig, Note, etc). But with your latest version, you can activate gate on/off from any page. Not sure if that's what you want - I personally prefer to be able to alter the gate sequence only in the Gate menu, so it's harder to accidentally alter the gate pattern while I'm trying to just change note values or gate lengths, etc.

I reloaded the official 0.1.39 firmware to confirm, because that's what I remember, and confirm that's the case.

**Comment by vafu** [comment]
@hotelbachelard yes, thanks! I've added this as my own preference but I accidentally added these changes to the firmware update, sorry about that, I'm going to remove that asap!

**Comment by hotelbachelard** [comment]
Hey @vafu - I found a lil bug in your 1.39: the Gate Probability bar doesn't show up in your version. :(
![20210227_212900](https://user-images.githubusercontent.com/26447094/109406087-40fd2100-7944-11eb-9666-a0f2346cdd31.jpg)
it's there in 1.40 (and the official 1.39)
![20210227_213108](https://user-images.githubusercontent.com/26447094/109406102-5b36ff00-7944-11eb-9c3e-3d729c0c7070.jpg)
